src/Workspaces/Core/Portable/Workspace/Solution/SourceGeneratedDocumentState.cs

```diff
@@ -168,13 +168,19 @@ public SourceGeneratedDocumentState WithGenerationDateTime(DateTime generationDa
         if (this.GenerationDateTime == generationDateTime)
             return this;
 
-        return Create(
-            Identity,
-            SourceText,
-            ParseOptions,
-            LanguageServices,
-            // We're just changing the date time only.  So the checksum will remain as is.
-            _lazyContentHash,
+        // Copy over all state as-is.  The generation time doesn't change any actual state (the same tree will be
+        // produced for example).
+        return new(
+            this.Identity,
+            this.LanguageServices,
+            this.Services,
+            this.Attributes,
+            this.ParseOptions,
+            this.TextAndVersionSource,
+            this.SourceText,
+            this.LoadTextOptions,
+            this.TreeSource!,
+            this._lazyContentHash,
             generationDateTime);
```

Comment:
previously, updating the generation-time would cause the doc to be reparsed.  there's no need for that at all.